### PR TITLE
Add support for stm32 bluepill

### PR DESCRIPTION
**Pull request by Javier-varez**

### CODE REVIEW

**Comment by ilg-ul** [comment]
Thank you for the patch, it seems that no functional changes were needed, only the board definition, right?

**Comment by Javier-varez** [comment]
Hi,

That's right. However I notice that there is a minor change in the microcontroller model (STM32F103C8T6), which has 64 kB of Flash instead of 128 kB for STM32F103RB. They share the same SVD file.

I have created the model for the MCU and updated the board to use it.

Best regards,
Javier Alvarez

File: hw/cortexm/stm32/mcus.c [hunk 1]
```diff
@@ -953,6 +953,19 @@ static const STM32PartInfo stm32_mcus[] = {
...
+        .name = TYPE_STM32F103C8, // STM32F103x[8B]
+        .cortexm = {
+            .flash_size_kb = 64,
+            .sram_size_kb = 20,
+
+            .svd_file_name = "STM32F103xx-qemu.json",
+            .svd_device_name = "STM32F103xx",
+        /**/
+        },
+        .stm32 = &stm32f103x8b,
```
Comment (ilg-ul) [inline hw/cortexm/stm32/mcus.c]: stm32f103x8b?

Comment (Javier-varez) [reply hw/cortexm/stm32/mcus.c]: Yes, they share the same capabilities. In fact they even share the datasheet. The difference is only flash size

Comment (ilg-ul) [reply hw/cortexm/stm32/mcus.c]: Ok

**Comment by ilg-ul** [comment]
I managed to update the build procedure for the new xPack Build Box v3.1.

In preparation for the next release, if you have no additions/changes to this PR, I'm ready to merge it.

Is it ok?

**Comment by Javier-varez** [comment]
Hi Liviu,

Yes, this is the latest revision, ready to merge. Thanks a lot!

BR,
Javier Álvarez

**Comment by ilg-ul** [comment]
Done, thank you. I'll check it tomorrow.

**Comment by ilg-ul** [comment]
The emulation if functional, but, if I read the schematics right, the led turns on when the GPIO pin is set to 0. 

And the colour is GREEN.

I also slightly changed the board name.

https://github.com/xpack-dev-tools/qemu/commit/9835ebb51f07935419e05ed5c8184bc50d0370a8

**Comment by Javier-varez** [comment]
That's strange. I have the board and the LED is actually red. I guess they may be manufactured with different LED colors depending on availability. Regarding if it's active low, you are completely right. I tested this with a blinky, but forgot to check polarity, sorry about that.

**Comment by ilg-ul** [comment]
Actually I did not find the authoritative documentation for the BluePill, only some examples where it was mentioned RED for power, GREEN for user.

Who designed it? I guess there are many chinese clones, but is there an official vendor?

**Comment by Javier-varez** [comment]
Yeah, that's a good point. I was never able to find an official vendor for this board. It probably originated from some Chinese company, but no one seems to claim ownership of the design.

As far as documentation goes, I was able to find this source, which seems to have a comprehensive amount of data about it:
https://stm32-base.org/boards/STM32F103C8T6-Blue-Pill

**Comment by ilg-ul** [comment]
https://www.youtube.com/watch?v=jLTi2YchcxA

shows a green led blinking.

**Pull request merged**